virocon/plotting.py

```diff
@@ -293,8 +317,138 @@ def plot_dependence_functions(model, semantics=None, par_rename={}, axes=None):
+                    x,
+                    density,
+                    color="#004488",
+                )
+            ax.set_xlabel(x_label)
+            ax.set_ylabel("probability density")
+            ax.set_title(f"n={len(data)}")
+
+            figures.append(fig)
+            axes_list.append(ax)
+
+        else:
+            # conditional
+            cond_dist = model.distributions[dim]
+            dist_per_interval = cond_dist.distributions_per_interval
+            n_intervals = len(dist_per_interval)
+
+            conditioning_idx = model.conditional_on[dim]
+            conditioning_symbol = semantics["symbols"][conditioning_idx]
+            conditioning_unit = semantics["units"][conditioning_idx]
+
+            slicer = model.interval_slicers[conditioning_idx]
+            interval_slices, conditioning_values, _ = slicer.slice_(
+                sample[:, conditioning_idx]
+            )
+            data_intervals = [sample[int_slice, dim] for int_slice in interval_slices]
+
+            # if the following fails, the sample was probably not used for fitting
+            # TODO raise proper exception then
+            np.testing.assert_allclose(
+                conditioning_values, cond_dist.conditioning_values
+            )
+            assert len(data_intervals) == len(cond_dist.data_intervals)
+            for i in range(len(data_intervals)):
+                np.testing.assert_allclose(
+                    np.sort(data_intervals[i]), np.sort(cond_dist.data_intervals[i])
+                )
+
+            fig, axes = _get_n_axes(n_intervals)
+            for interval_idx in range(n_intervals):
+                cond_val = conditioning_values[interval_idx]
+                data = data_intervals[interval_idx]
+                x = np.linspace(np.min(data), np.max(data))
+                dist = dist_per_interval[interval_idx]
+                ax = axes[interval_idx]
+                ax.hist(
+                    data,
+                    bins="doane",
+                    density=True,
+                    color="#000000",
+                    histtype="stepfilled",
+                    alpha=0.2,
+                )
+                if plot_pdf:
+                    density = dist.pdf(x)
+                    ax.plot(
+                        x,
+                        density,
+                        color="#004488",
+                    )
+                ax.set_xlabel(x_label)
+                ax.set_ylabel("probability density")
+                title = f"{conditioning_symbol} = {cond_val:.3f} {conditioning_unit}, n={len(data)}"
+                ax.set_title(title)
+
+            figures.append(fig)
+            axes_list.append(axes)
+
+    return figures, axes_list
+
+
 def plot_2D_isodensity(
-    model, sample, semantics=None, swap_axis=False, limits=None, levels=None, ax=None
+    model,
+    sample,
+    semantics=None,
+    swap_axis=False,
+    limits=None,
+    levels=None,
+    n_grid_steps=250,
```

Comment:
Why not simply add the new keyword in last position to avoid the breaking change?

I know that we currently have "ax" as the last keyword in the plot functions, however, I think that is not an important convention. If users remember the keyword order of the plot functions, I think they likely only remember the first two keywords.